api/src/main/java/org/apache/iceberg/util/ByteBuffers.java

```diff
@@ -46,6 +47,15 @@
     }
   }
 
+  public static ByteBuffer reuse(ByteBuffer reuse, int length) {
+    Preconditions.checkArgument(reuse.hasArray() && reuse.arrayOffset() == 0 && reuse.capacity() == length,
+        "Cannot reuse buffer: Should be an array %s, should have an offset of 0 %s, should be of size %s was %s",
+        reuse.hasArray(), reuse.arrayOffset(), length, reuse.capacity());
```

Comment:
Nit: I'd prefer to split this into three `Preconditions.checkArgument` because we can easily find out what's the reason that the conditions fail.

Reply:
Agree on this